### ISSUE TITLE
Better default bindings for Vive Wands

### ISSUE THREAD

**Issue by praydog** [praydog/REFramework#98]
Current bindings can be found here: https://github.com/praydog/REFramework/blob/master/src/mods/vr/Bindings.cpp#L414

**Comment by KasaiMonkey** [comment]
Hi praydog,
been playing around with the bindings for vive wands.
So far i've been able to play with no issues by assigning a extra click action for the trigger, setting it as abutton.

Only thing I haven't been able to figure out is the dial weapon change for the vive, though i've just been going to the inventory to change weapons anyways.

**Comment by draknes89** [comment]
HI Praydog!

i have a problem with my vive controllers setup heve any chance  to get a full vive setup guide??? im so dumb i know and sorry for my english.... i dont know how to run in re3 and always die in the first 5 min.... pls help

**Comment by Schoober** [comment]
> 

How abouts did you make this change? I'm unable to interact with anything, but i CAN shoot

**Comment by Schoober** [comment]
vive has no abutton binding

**Comment by khyrall** [comment]
Here is how I set up my bindings. It works for all issues I have seen in the discord (A button hold, B Button, LB working) and uses DPAD instead of default's first person/third person switch. I shared it as a community binding also if anyone wants to try it out/give feedback. 

![RE2Bindings](https://user-images.githubusercontent.com/54683001/147895783-6d4fc2a0-1a90-42d5-942f-133c32fedb41.png)

**Comment by draknes89** [comment]
Thx man

khyrall ***@***.***> ezt írta (időpont: 2022. jan. 3., H
3:17):

**Comment by draknes89** [comment]
> Here is how I set up my bindings. It works for all issues I have seen in the discord (A button hold, B Button, LB working) and uses DPAD instead of default's first person/third person switch. I shared it as a community binding also if anyone wants to try it out/give feedback.
> 
> ![RE2Bindings](https://user-images.githubusercontent.com/54683001/147895783-6d4fc2a0-1a90-42d5-942f-133c32fedb41.png)

And how you run??? this is my big problem!!!!

**Comment by khyrall** [comment]
> > Here is how I set up my bindings. It works for all issues I have seen in the discord (A button hold, B Button, LB working) and uses DPAD instead of default's first person/third person switch. I shared it as a community binding also if anyone wants to try it out/give feedback.
> > ![RE2Bindings](https://user-images.githubusercontent.com/54683001/147895783-6d4fc2a0-1a90-42d5-942f-133c32fedb41.png)
> 
> And how you run??? this is my big problem!!!!

Left Trackpad Click does running for me.

**Comment by draknes89** [comment]
thx!!

**Comment by khyrall** [comment]
Weird, that page is where community bindings show up for me. You may have to create your own, if you want to model off mine make sure to turn mirror mode off when creating the bindings.

**Comment by khyrall** [comment]
Yeah, that's because the actions don't match up 1 for 1 with controller buttons. Each action has a corresponding button when in left or right. 

Here are some I have tested out:
grip 
 - left does LB. 
 - right readies weapon
abutton 
 - left wand seems to do B
 - right wand seems to do A
bbutton
 - left wand seems to open inventory
 - right wand seems to do X
trigger
 - left wands seems to do nothing
 - right wand shoots if weapon readied

**Comment by GT500org** [comment]
I've been messing with the binds for Vive Wands in RE2 with this mod today, and I can't figure out how to get the right trigger to fire the gun. Even the guns with infinite ammo won't fire. I've managed to get most of the other binds to the point where they work (thanks mostly to @khyrall and the screenshot of his binds), but shooting the gun still eludes me. It doesn't help that the SteamVR controller binds interface eventually dies on me, and I have to restart SteamVR to get it working again...

BTW: Is it possible to make the knife toggleable instead of requiring to hold the button? It's extremely difficult to use the touchpad on a Vive Wand to walk while holding down the grip button.

@praydog awesome mod BTW.

**Comment by khyrall** [comment]
> I've been messing with the binds for Vive Wands in RE2 with this mod today, and I can't figure out how to get the right trigger to fire the gun. Even the guns with infinite ammo won't fire. I've managed to get most of the other binds to the point where they work (thanks mostly to @khyrall and the screenshot of his binds), but shooting the gun still eludes me. It doesn't help that the SteamVR controller binds interface eventually dies on me, and I have to restart SteamVR to get it working again...
> 
> BTW: Is it possible to make the knife toggleable instead of requiring to hold the button? It's extremely difficult to use the touchpad on a Vive Wand to walk while holding down the grip button.
> 
> @praydog awesome mod BTW.

Thanks for the callout, glad it helped out! For shooting, are you pressing the right grip while pulling the trigger? (If using my binds, otherwise whatever grip is assigned to for your right wand.) RE2 makes you aim before firing so holding the right grip is needed to "aim" the gun. 

I haven't tested it out, but its possible to setup bindings as toggles instead of buttons. So theoretically you could change the left grip to toggle and it might have the effect you want, but I haven't tested it out.

**Comment by GT500org** [comment]
> For shooting, are you pressing the right grip while pulling the trigger? (If using my binds, otherwise whatever grip is assigned to for your right wand.) RE2 makes you aim before firing so holding the right grip is needed to "aim" the gun. You'll know its ready to fire if the ammo count is appearing in the lower right.

Thanks, that was the problem (not pressing the grip). It had been a while since I had played the game on flatscreen, and I had forgotten that you had to hold a button to be able to shoot.

**Comment by khyrall** [comment]
I recently had more time to play and learned that this binding does not allow you to defend with secondary weapons once grabbed by a zombie. My guess is that this is what the trigger binding does on the left wand. I'll try playing with bindings this weekend and see if I can get an updated version.

**Comment by khyrall** [comment]
I have published an updated community binding that allows you to defend with your secondary when grabbed by a zombie.
Name: Updated Mapping by Khyrall
Description:
| Left Grip - LB (Ready Secondary)
| Left Menu - Once for Y (open Inventory). Twice for B
| Left Trigger - LB (Defend from Zombie)
| Left Trackpad - Movement. Click for sprint.
| Right Grip - Ready Weapon
| Right Menu - Once for A. Twice for X (reload). Holding A works but brings up a steam icon.
| Right Trackpad - DPAD (Weapon Dial) Directions are set as clicks.
![RE2UpdatedBindings](https://user-images.githubusercontent.com/54683001/148648780-aeb50e4c-6c5b-49a9-8636-6f6107feaa5f.png)
[re2BindingsJSON.txt](https://github.com/praydog/REFramework/files/7833407/re2BindingsJSON.txt)

**Comment by Clodo76** [comment]
Hi, i'm playing RE8 with Valve Index (and Pimax), and some buttons (like map) sound missing (no issue with RE2 or RE3, only with RE8)
I try to edit steamvr binding, but if RE8 is running something break (stuck on "loading profiles") and i can't reach the mapping editor of screenshot above.
If RE8 is not running, i cannot switch the application to edit (steamvr menu list only VR games, the games listed in SteamVR Home panels)
Any advice? Thx

**Comment by khyrall** [comment]
Has anyone gotten weapon switching to work with RE7 with Vive? The Dpad actions mapped in the examples above don't seem to be associated with the dpad buttons in RE7 from what I've tested.

**Comment by khyrall** [comment]
> Has anyone gotten weapon switching to work with RE7 with Vive? The Dpad actions mapped in the examples above don't seem to be associated with the dpad buttons in RE7 from what I've tested.

I managed to get some bindings working for RE7, played up to halfway through old house no issues. D-pad mappings are double presses/long presses on some buttons but they are all doable. 

![RE7Bindings](https://user-images.githubusercontent.com/54683001/169546913-e4badef6-e3a0-4bf1-bbb8-c770b28f345e.png)

**Comment by KateHanami** [comment]
here's good (not idea) bidings I found
![image](https://user-images.githubusercontent.com/70842044/175779155-c7f7476a-fffc-4822-a1a8-943af775bdde.png)
I would prefer to have a button to use the dpad shortcuts instead oof spreading them into random buttons

**Comment by Alainx277** [comment]
> Yeah, that's because the actions don't match up 1 for 1 with controller buttons. Each action has a corresponding button when in left or right.
> 
> Here are some I have tested out:
> 
> grip
> 
> * left does LB.
> * right readies weapon
> 
> abutton
> 
> * left wand seems to do B
> * right wand seems to do A
> 
> bbutton
> 
> * left wand seems to open inventory
> * right wand seems to do X
> 
> trigger
> 
> * left wands seems to do nothing
> * right wand shoots if weapon readied

Unfortunately I missed this comment and spent hours trying to get the DPAD working.

It seems that binding dpad on the right controller just doesn't work. But the game is playable without it.

I recommend binding left grip to quick weapon switch for RE8.

**Comment by Alex-Young1113** [comment]
hi did anybody figure out how to input dpad using quest3/VD?
